src/objective-c/GRPCClient/private/GRPCWrappedCall.m

```diff
@@ -0,0 +1,193 @@
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#import <Foundation/Foundation.h>
+#import "GRPCWrappedCall.h"
+#import "GRPCCompletionQueue.h"
+#import "NSDictionary+GRPC.h"
+#import "NSData+GRPC.h"
+#import "NSError+GRPC.h"
+#include <grpc/grpc.h>
+#include <grpc/byte_buffer.h>
+#include <grpc/support/alloc.h>
+
+@implementation GRPCWrappedCall{
+  grpc_call *call;
+  GRPCCompletionQueue *queue;
+}
+
+- (instancetype)init {
+  return [self initWithChannel:nil method:nil host:nil];
+}
+
+- (instancetype)initWithChannel:(GRPCChannel *)channel method:(NSString *)method host:(NSString *)host {
+  if (!channel || !method || !host) {
+    [NSException raise:NSInvalidArgumentException format:@"channel, method, and host cannot be nil."];
+  }
+  
+  if (self = [super init]) {
+    static dispatch_once_t initialization;
+    dispatch_once(&initialization, ^{
+      grpc_init();
+    });
+    
+    const char *method_str = [method UTF8String];
+    const char *host_str = [host UTF8String];
+    queue = [GRPCCompletionQueue completionQueue];
+    call = grpc_channel_create_call(channel.unmanagedChannel, queue.unmanagedQueue, method_str, host_str, gpr_inf_future);
+    if (call == NULL) {
+      return nil;
+    }
+  }
+  return self;
+}
+
+- (void)startBatch:(NSDictionary *)ops handleCompletion:(GRPCCompletionHandler)handleCompletion {
+  size_t nops = ops.count;
+  grpc_op *ops_array = gpr_malloc(nops * sizeof(grpc_op));
+  size_t index = 0;
+  NSMutableDictionary * __block opProcessors = [NSMutableDictionary new];
+  
+  grpc_metadata *send_metadata = NULL;
+  grpc_metadata_array *recv_initial_metadata;
+  grpc_metadata_array *recv_trailing_metadata;
+  grpc_byte_buffer *send_message;
+  grpc_byte_buffer **recv_message = NULL;
+  grpc_status_code *status_code;
+  char **status_details;
+  size_t *status_details_capacity;
+  for (id key in ops) {
+    id (^opBlock)(void);
+    grpc_op *current = &ops_array[index];
+    switch ([key intValue]) {
+      case GRPC_OP_SEND_INITIAL_METADATA:
+        current->data.send_initial_metadata.count = [ops[key] grpc_toMetadataArray:&send_metadata];
+        current->data.send_initial_metadata.metadata = send_metadata;
+        opBlock = ^{
+          gpr_free(send_metadata);
+          return @YES;
+        };
+        break;
+      case GRPC_OP_SEND_MESSAGE:
+        send_message = [ops[key] grpc_byteBuffer];
```

Comment:
The problem is that the tag doesn't *just* indicate the type. If the tag is `GRPC_OP_SEND_CLOSE_FROM_CLIENT` or any of `GRPC_OP_RECV_*`, the value isn't even read, because the tag is just an indication that the action should be performed.